scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,567 @@
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+# Define the special kubectl
+k() {
+    KUBECONFIG="$kubeconfig" kubectl $*
+}
+
+
+###############################################################################
+# Script info
+###############################################################################
+echov -e "${COLOR_INFO}$($0 --version)${COLOR_NONE}\n"
+echov -e "Script run on ${COLOR_INFO}$timestamp${COLOR_NONE}"
+
+
+###############################################################################
+# Check dependencies
+###############################################################################
+echov "Checking script dependencies:"
+
+is_installed 'kubectl' \
+    'kubectl' \
+    'kubectl version --client | head -1 | cut -d" " -f3'
+
+is_installed 'helm' \
+    'helm' \
+    'helm version | cut -d"\"" -f2'
+
+is_installed 'jq' \
+    'jq' \
+    'jq --version'
+
+
+###############################################################################
+# Ensure output folder
+###############################################################################
+if [[ "$arg_logs" == "true" || "$arg_yaml" == "true" ]] ; then
+    yes "yes" | rm -vRI "/tmp/kubestellar-snapshot" > /dev/null 2>&1 || true
+    mkdir -p "$output_folder"
+fi
+
+
+###############################################################################
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/k/generated/k_config/
+###############################################################################
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    kubeconfig="$arg_kubeconfig"
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    kubeconfig="$KUBECONFIG"
+else
+    kubeconfig="$HOME/.kube/config"
+fi
+echov -e "Using kubeconfig(s): ${COLOR_INFO}$kubeconfig${COLOR_NONE}"
+
+
+###############################################################################
+# Determine the list of contexts to search
+###############################################################################
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(k config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+
+echov "Validating contexts(s): "
+vc_n=0
+valid_context=()
+for context in "${contexts[@]}" ; do # for all contexts
+    [[ -z "$context" ]] && continue
+    if k --context $context get secrets -A > /dev/null 2>&1 ; then
+        echov -e "${COLOR_GREEN}\xE2\x9C\x94${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE}"
+        valid_context[vc_n]="$context"
+        vc_n=$((vc_n+1))
+    else
+        echov -e "${COLOR_RED}X${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE}"
+    fi
+done
+contexts=("${valid_context[@]}")
+
+if [[ -z "${contexts[@]}" ]] ; then
+    echoerr "no context(s) found in the kubeconfig file $kubeconfig!"
+    exit 1
+fi
+
+
+###############################################################################
+# Look for KubeStellar Helm chart
```

Comment:
If this script had documentation then it would mention that the script only works if KubeStellar was installed via the core Helm chart.

Reply:
I'll add a mention but there is no other way to install Kubestellar, right?

Reply:
```shell
Note 1: This script expects that KubeStellar was installed with its Helm chart and
        to find the latter in the host or provided kubeconfig(s) and context(s).
```

Reply:
We do not yet support any other way but I suspect that we might in the future. For example, Issue #2577 is someone asking for general setup instructions and we have not yet proven that the core Helm chart will work for them. For another example, recall the ambition to be able to add KubeStellar to a pre-existing OCM system without any KubeFlex already being there nor being added; that would differ more than a little from the current pattern (as well as require enabling changes in the controller codes).

Reply:
I'm not sure that even in that case will will actually not adapt to use the helm chart.
Furthermore, I'm not sure how the resources structure will look like so I suggest that we revise the script when we know more.

Reply:
Indeed, it would be premature to make the script do anything else now. My point here is only that this specificity should be documented.

Reply:
I think it is